### ISSUE TITLE
Erro Database Provider with Evolution v2

### ISSUE THREAD

**Issue by brauliodiasribeiro** [EvolutionAPI/evolution-api#1024]
### Welcome!

- [X] Yes, I have searched for similar issues on [GitHub](https://github.com/EvolutionAPI/evolution-api/issues) and found none.

### What did you do?

Evolution v2 don't work with database provider = true

### What did you expect?

.

### What did you observe instead of what you expected?

.

### Screenshots/Videos

![381316541-49041269-a04a-4471-9163-5e2d660368d4](https://github.com/user-attachments/assets/893089f1-2dd0-4da8-af6e-f6bcc892ad8c)
The name of my postgres installation is "Postgres" with a uppercase letter and my database is "evolutionapi"

![381317008-c4baed62-9b09-49d9-a3ca-8492966ff001](https://github.com/user-attachments/assets/c03c7145-1026-4511-becd-179b41cbb964)

### Which version of the API are you using?

v2.1.2

### What is your environment?

Docker

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I've updated the `src/config/env.config.ts` file to ensure that the database connection string is correctly read from the environment variables. This change allows for proper capitalization of the Postgres host name and database name.

I also created pull request: https://github.com/EvolutionAPI/evolution-api/pull/1025


> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission

**Comment by brauliodiasribeiro** [comment]
Don't work

**Comment by bryanwac** [comment]
Estou passando pelo mesmo problema e tentando corrigir.
O principal problema é que falta uma variável no env do docker que é:

`DATABASE_PROVIDER= <nome do seu banco aqui>
`
Exemplo: `DATABASE_PROVIDER=postgresql`

EU utilizo apenas o armazenamento local, e entendi que o problema real é que a variável:
`DATABASE_ENABLED=false`

Não está sendo respeitada, mesmo eu tendo mostrado que quero utilizar o storage local apenas, ele ainda tenta conectar ao banco que é inexistente e não tinha provider.


Ou seja, defina o `DATABASE_PROVIDER` no env do seu docker, para o nome do banco que você utiliza e irá funcionar corretamente.

**Comment by bryanwac** [comment]
Outra solução:

Pare de usar: atendai/evolution-api:latest
Use: atendai/evolution-api:v1.8.2

Dessa forma seu docker nao vai dar pull sempre na ultima imagem e vc nao vai ter problemas envolvendo seu ENV pois ele só vai mudar quando vc mudar a imagem manualmente

Ps: This will make ur api stop throwing the error Database provider invalid too, cz this version do not expect a `DATABASE_PROVIDER `on docker env

**Comment by Lucas-Alves-Silva** [comment]
Defini essa variável DATABASE_PROVIDER=postgresql no arquivo .env do docker, porém, continuo tendo o mesmo erro.
Estou horas tentando resolver isso.
Alguma outra dica?
Setei essa imagem atendai/evolution-api:v2.2.0

**Comment by bryanwac** [comment]
> Defini essa variável DATABASE_PROVIDER=postgresql no arquivo .env do docker, porém, continuo tendo o mesmo erro. Estou horas tentando resolver isso. Alguma outra dica? Setei essa imagem atendai/evolution-api:v2.2.0

O env pra 2.2.0 é diferente do env pra 1.8.
Verifica na doc oficial quais variáveis mudaram e como definir elas corretamente.
A 2.0 define o banco de outra forma

**Comment by brauliodiasribeiro** [comment]
> Defini essa variável DATABASE_PROVIDER=postgresql no arquivo .env do docker, porém, continuo tendo o mesmo erro. Estou horas tentando resolver isso. Alguma outra dica? Setei essa imagem atendai/evolution-api:v2.2.0

#PROVIDER_ENABLED=false
DATABASE_ENABLED=true
DATABASE_PROVIDER=postgresql
DATABASE_CONNECTION_URI=postgresql://YOUR-USER:YOUR-PASS@postgres:5432/evolutionapi?schema=public
DATABASE_CONNECTION_CLIENT_NAME=evolution_v2

**Comment by dpaes** [comment]
Certo, eu fiz ja diversas instalações aqui e nunca tive esse problema, a não ser que vc esteja colocando o postgres em uma versão incompatível para o evolution. Se você usar um postgres muito antigo o evolution também não reconhecerá ele. A versão que sempre uso é a 16.4 do postgres.

**Comment by dpaes** [comment]
outros problemas podem ser de rede, postgres em uma rede diferente do evolution, pode fazer com que o evolution não consiga acessar o postgres. Usando docker, bastando colocar o evolution e o postgres na mesma rede, o evolution consegue acessar a porta interna do postgres sem precisar expor nenhuma porta no postgres. Do contrario você precisaria expor a porta do postgres para a internet.

**Comment by brauliodiasribeiro** [comment]
@dpaes eu nao consigo identificar o problema.

Eu uso servidor Unraid, e os dockers estao na mesma rede. Utilzo o postgres:16.
Todos os outros docker que tenho trabalhando em conjunto funcionam.

![image](https://github.com/user-attachments/assets/b4e987ba-24b0-45e6-a3f3-0890ee0457bf)

**Comment by dpaes** [comment]
ja tentou testar em outro provider ou uma outra distro linux? @brauliodiasribeiro, pq realmente nunca tive esse problema, sempre configurei isso e tudo deu certo por aqui, ja fiz dezenas de instalações tanto v1 quanto v2 do evolution

**Comment by brauliodiasribeiro** [comment]
> ja tentou testar em outro provider ou uma outra distro linux? @brauliodiasribeiro, pq realmente nunca tive esse problema, sempre configurei isso e tudo deu certo por aqui, ja fiz dezenas de instalações tanto v1 quanto v2 do evolution

Nao tenho como tentar em outra distro pq Unraid é um sistema pra servidor que utilizo proffisonalmente.
Ele é baseado em distribuicao Linux Slackware

**Comment by brauliodiasribeiro** [comment]
Alguem conseguiu resolver o problema ?
Eu sei que o problema esta relacionado ao arquivo .env
O docker nao esta conseguindo ler ele. Achei que era problema de permissao, mas nao funcionou.
Alguem sabe resolver essa questao .env?

Has anyone managed to solve the problem?
I know the issue is related to the .env file.
Docker is not able to read it. I thought it was a permission issue, but it didn't work.
Does anyone know how to resolve this .env issue?

**Comment by dpaes** [comment]
@brauliodiasribeiro, faça o teste em outro provider e pegue um servidor de testes so pra vc atestar o que disse. Provavelmente tenha algo diferente nesse Unraid que por algum motivo não funcione corretamente. Se em outro provider funcionar, a unica questão é descobrir o que que o Unraid não tem que nos outros lugares tem.

**Comment by brauliodiasribeiro** [comment]
> [@brauliodiasribeiro](https://github.com/brauliodiasribeiro), faça o teste em outro provider e pegue um servidor de testes so pra vc atestar o que disse. Provavelmente tenha algo diferente nesse Unraid que por algum motivo não funcione corretamente. Se em outro provider funcionar, a unica questão é descobrir o que que o Unraid não tem que nos outros lugares tem.

Oiii isso que voce disse foi o que eu fiz. Em outro funcionam, o X ta no Unraid. Mas talvez seja algo de permissoes, pois eu tenho uns 20 dockers nele (foi feito pra isso), so o Evolution nao funciona.

**Comment by dpaes** [comment]
pois é @brauliodiasribeiro, oq vc pode fazer é usar uma vps pequena em outro provider pra funcionar o evolution de forma externa ao Unraid, porém o resto estando la né, ja que não foi identificado a raiz do problema nele.

**Comment by LucianTavares** [comment]
Quem estiver com o problema de `Invalid Provider` eu fiz o seguinte para resolver:

No `docker-compose.yaml` defini as variáveis da seguinte forma:
```
    environment:
      - DATABASE_ENABLED=true
      - DATABASE_CONNECTION_TYPE=postgres
      - DATABASE_CONNECTION_URI=postgres://chatwoot:chatwoot@postgres:5432/evolution_db
      - DATABASE_SSL_CERT=false
      - DATABASE_PROVIDER=postgresql
```

Veja que `DATABASE_CONNECTION_URI` é diferente de `DATABASE_URL`
